### ISSUE TITLE
Whitespace issue with BM font labels when replacing strings.

### ISSUE THREAD

**Issue by slembcke** [cocos2d/cocos2d-objc#1018]
http://forum.cocos2d-swift.org/t/cclabelbmfont-bug-with-newlines/15408

**Comment by iMobCoding** [comment]
While you're at it, it should be very useful to have an option to alter space between new lines programmatically...

**Comment by glazedkoala** [comment]
I've posted a fix to this in the original cocos2d forum post.

**Comment by slembcke** [comment]
Yes I saw! I've been hunting bugs this week, and wasn't sure if I would get to this one or not. Will take a look.

Thanks for helping look into this!

**Comment by tri-nomad** [comment]
Hi Scott
Created a pull request. Although it contains fix for 2 different issues, I am using same issue#.

Just to clarify about _width being ignored.
-- If I have a CCLabelBMFont and I am updating it such that appending single character asynchronously (say using keyboard) then it tends to ignore Width completely. 

-- I took liberty to add patch by @glazedkoala (rearranged if-else)

Thanks!
